types/passport/index.d.ts

```diff
@@ -6,20 +6,24 @@
 //                 Tomek Łaziuk <https://github.com/tlaziuk>
 //                 Daniel Perez Alvarez <https://github.com/danielpa9708>
 //                 Kevin Stiehl <https://github.com/kstiehl>
+//                 Oleg Vaskevich <https://github.com/vaskevich>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
 declare global {
     namespace Express {
+        interface AuthInfo {}
+        interface User {}
+
         interface Request {
-            authInfo?: any;
-            user?: any;
+            authInfo?: AuthInfo;
```

Comment:
https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes
This shouldnt be interface without properties. It needs to be either object or any

Reply:
As mentioned in the PR description, this is very much intentional. The `User` and `AuthInfo` properties can be augmented by users depending on how Express is used, and this can only be done in a type safe matter with an interface. If it's `any` or an `object`, it cannot be extended. For more details, see: https://www.typescriptlang.org/docs/handbook/declaration-merging.html

For our use case, this is leading to type conflicts because we're setting `user` to have an explicit type when using Passport, and the current declaration of `any` is conflicting with it, because it does not support declaration merging. IMO a third-party package updating Express type declarations should play nicely with it, not be greedy with common properties like `user` and `authInfo` by setting it them to something that cannot be extended.